app/src/main/java/com/github/ashutoshgngwr/noice/receiver/BootCompletedReceiver.kt

```diff
@@ -0,0 +1,28 @@
+package com.github.ashutoshgngwr.noice.receiver
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import com.github.ashutoshgngwr.noice.repository.AlarmRepository
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class BootCompletedReceiver : BroadcastReceiver() {
+
+  @set:Inject
+  internal lateinit var alarmRepository: AlarmRepository
+
+  override fun onReceive(context: Context?, intent: Intent?) {
+    if (Intent.ACTION_BOOT_COMPLETED != intent?.action && "android.intent.action.QUICKBOOT_POWERON" != intent?.action) {
+      return
+    }
+
+    runBlocking {
+      withContext(Dispatchers.IO) { alarmRepository.rescheduleAll() }
```

Comment:
It may be possible that we don't have permission to schedule exact alarms after reboot.